### PR TITLE
fix: inform user when default connection is not in line with machine native option

### LINKED ISSUE #7052: Rootful podman machine gets stuck using rootless connection after another rootless machine is deleted
### Bug description

It seems that in some situations a rootful podman machine can get stuck using a rootless connection if the user deletes another podman machine which is rootless and is set to default at the same time.

Additionally it seems that this scenario also makes the "This machine is not set to default" prompt to be displayed at all until such time that the user manually shuts down Podman Desktop and reopens it.

### Operating system

Windows 11

### Installation Method

Installer from website/GitHub releases

### Version

1.10.0

### Steps to reproduce

1. Have two podman WSL machines available on the system, one rootful and one rootless, the rootless machine should be set to default and both machines should be running at the same time.
2. Go to Settings -> Resources menu.
3. Stop the rootless podman machine.
4. A popup will be triggered during the stop action which will notify the user that the rootful machine is not set to default, select the `Ignore` option on the popup.
5. After the rootless podman machine finishes shutting down, delete it.
6. At this point the bug should have reproduced, going into CLI running the command `podman system connection list` will display the rootful podman machines's connections, but will show that the rootless connection is the one set to default, not the rootful one as it should be.
7. Additionally if the user makes anymore podman machines from PD, either rootful or rootless the "machine is not set as default" prompt will no longer be displayed until the user completely closes PD and reopens it.

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

This PR adds a check from the desktop side to see if someone is running a machine using the wrong connection. Like the running machine is rootful but the default connection is rootless.

There is a known bug on podman https://github.com/containers/podman/issues/22993 that it happens when you delete a machine but it also may happen if the user mess up using the cli. 
This way we prompt a message to inform the user that they should update the connection

### Screenshot / video of UI

update_connection

### What issues does this PR fix or reference?

It resolves https://github.com/containers/podman-desktop/issues/7052 and maybe https://github.com/containers/podman-desktop/issues/6455

### How to test this PR?

1. Create a rootful machine and start it
2. Create another rootful/rootless (it's the same) machine and start it
3. At this point, on Podman Desktop, you will be asked if you want to update the default connection -> yes
4. Stop the latest rootful/rootless machine -> on Podman Desktop you will be asked if you want to update the default connection -> yes -> the rootful connection of the first machine is set to default
5. delete the latest rootful/rootless machine.
6. if you check again you will see the default connection of the first machine has been updated to the rootless one
7. a message is prompted to update the connection

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: extensions/podman/src/extension.ts [hunk 1]
```diff
@@ -367,6 +368,39 @@ export async function checkDefaultMachine(machines: MachineJSON[]): Promise<void
     }
   }
 
+  // check if connection is in sync with machine. If the default connection is rootless but the machine is rootful ask the user to update the connection
+  if (defaultConnectionNotify && !!runningMachine?.Default) {
+    const defaultConnection = await getDefaultConnection();
+    const isRootful = await isRootfulMachine(runningMachine.Name);
+    if (!defaultConnection?.Name.endsWith(ROOTFUL_SUFFIX) && isRootful) {
```
Comment (jeffmaury) [inline extensions/podman/src/extension.ts:375]: I don't get it you are comparing the default connection with a non default machine ?

Comment (lstocchi) [reply extensions/podman/src/extension.ts:375]: In the if above i check that the running machine is the default one

**Review by jeffmaury — APPROVED** [review]
LGTM

**Comment by lstocchi** [comment]
Rebasing as i saw one gh workflow didn't trigger

**Comment by lstocchi** [comment]
@benoitf are you good with this change?

**Review by benoitf — APPROVED** [review]
@lstocchi yes

**Pull request merged**